src/server/game/Entities/Player/Player.cpp

```diff
@@ -27553,3 +27605,228 @@ void Player::SummonPet(uint32 entry, float x, float y, float z, float ang, PetTy
+            return true;
+
+        if ((movementInfo.GetMovementFlags() & MOVEMENTFLAG_ONTRANSPORT) || HasUnitMovementFlag(MOVEMENTFLAG_ONTRANSPORT))
+            return true;
+
+        if (UnderACKmount())
+            return true;
+
+        if (IsSkipOnePacketForASH())
+        {
+            SetSkipOnePacketForASH(false);
+            return true;
+        }
+
+        Position npos = movementInfo.pos;
+        if (sWorld->getBoolConfig(CONFIG_ANTICHEAT_IGNORE_CONTROL_MOVEMENT_ENABLED))
+        {
+            if (HasUnitState(UNIT_STATE_ROOT) && !UnderACKRootUpd())
+            {
+                bool unrestricted = npos.GetPositionX() != GetPositionX() || npos.GetPositionY() != GetPositionY();
+                if (unrestricted)
+                {
+                    sLog->outString("CheckMovementInfo :  Ignore control Hack detected for Account id : %u, Player %s", GetSession()->GetAccountId(), GetName().c_str());
+                    sWorld->SendGMText(LANG_GM_ANNOUNCE_MOVE_UNDER_CONTROL, GetSession()->GetAccountId(), GetName().c_str());
+                    return false;
+                }
+            }
+        }
+
+        if (HasUnitState(UNIT_STATE_IGNORE_ANTISPEEDHACK))
+            return true;
+
+        float distance, movetime, speed, difftime, normaldistance, delay, delaysentrecieve, x, y;
+        distance = npos.GetExactDist2d(this);
+
+        if (!jump && !CanFly() && !isSwimming())
+        {
+            float diffz = fabs(movementInfo.pos.GetPositionZ() - GetPositionZ());
+            float tanangle = distance / diffz;
+
+            if (movementInfo.pos.GetPositionZ() > GetPositionZ() &&
+                diffz > 1.87f &&
+                tanangle < 0.57735026919f) // 30 degrees
+            {
+                std::string mapname = GetMap()->GetMapName();
+                sLog->outString("CheckMovementInfo :  Climb-Hack detected for Account id : %u, Player %s, diffZ = %f, distance = %f, angle = %f, Map = %s, mapId = %u, X = %f, Y = %f, Z = %f",
+                    GetSession()->GetAccountId(), GetName().c_str(), diffz, distance, tanangle, mapname.c_str(), GetMapId(), GetPositionX(), GetPositionY(), GetPositionZ());
+                sWorld->SendGMText(LANG_GM_ANNOUNCE_WALLCLIMB, GetSession()->GetAccountId(), GetName().c_str(), diffz, distance, tanangle, mapname.c_str(), GetMapId(), GetPositionX(), GetPositionY(), GetPositionZ());
+                return false;
+            }
+        }
+
+        uint32 oldstime = GetLastMoveServerTimestamp();
+        uint32 stime = getMSTime();
+        uint32 ping, latency;
+        movetime = movementInfo.time;
+        latency = GetSession()->GetLatency();
+        ping = std::max(uint32(60), latency);
+
+        speed = GetSpeed(MOVE_RUN);
+        if (isSwimming())
+            speed = GetSpeed(MOVE_SWIM);
+        if (IsFlying() || this->CanFly())
+            speed = GetSpeed(MOVE_FLIGHT);
+
+        delaysentrecieve = (ctime - oldstime) / 10000000000;
```

Comment:
`ctime` and `oldstime` are integers, but their division is used in a floating point context (`delaysentrecieve`), probably is not a big deal but there is a chance to precision loss